libpod/image/pull.go

```diff
@@ -198,12 +200,74 @@ func (ir *Runtime) pullGoalFromImageReference(ctx context.Context, srcRef types.
 
 // pullImageFromHeuristicSource pulls an image based on inputName, which is heuristically parsed and may involve configured registries.
 // Use pullImageFromReference if the source is known precisely.
-func (ir *Runtime) pullImageFromHeuristicSource(ctx context.Context, inputName string, writer io.Writer, authfile, signaturePolicyPath string, signingOptions SigningOptions, dockerOptions *DockerRegistryOptions, label *string) ([]string, error) {
+func (ir *Runtime) pullImageFromHeuristicSource(ctx context.Context, config *NewImageConfig) ([]string, error) {
```

Comment:
So, is it, after all, a `NewImageConfig` for `ImageRuntime().New()`, or really a `PullConfig` for `pullImage*`?

(I suppose that strongly depends on the location of the `pullAlways`/`update` options.)